app/models/compute_resources/foreman/model/ovirt.rb

```diff
@@ -695,5 +696,27 @@ def set_vm_interfaces_attributes(vm, vm_attrs)
       end
       vm_attrs
     end
+
+    def set_vm_volumes_attributes(vm, vm_attrs)
+      template = template(vm_attrs[:template]) if vm_attrs[:template]
+      template_volumes_name_id = {}
+      template&.volumes&.each { |volume| template_volumes_name_id[volume.name] = volume.id }
```

Comment:
This part might deserve it's own method, it's a lot of logic in a single line and makes it a little hard to read when it's here.